### DISCUSSION TITLE
Problems with States/Custom Images

### DISCUSSION THREAD

**Discussion by schweito** [nguyenquyhy/Flight-Tracker-StreamDeck#319] [Q&A]
Dear Comunity,

I am new to the Stream Desk and new to to Flight Tracker plugin - however, I did watch some videos and read the examples. I wanted test a simple Taxi Light toggle function and even though the toggle event works fine, my custom image is not changing as the state of the toggle in the simulator changes. The .json-file generated by the flighttracker-plugin looks like this:

{
  "Actions": {
    "7,0": {
      "Name": "Generic Toggle Button",
      "Settings": {
        "FeedbackValue": "LIGHT TAXI",
        "ImageOff": "C:/Users/korre/Pictures/EMB-110/Resized/PNG/TAXI_OFF.png",
        "ImageOff_base64": null,
        "ImageOn": "C:/Users/korre/Pictures/EMB-110/Resized/PNG/TAXI_ON.png",
        "ImageOn_base64": null,
        "ToggleValue": "TOGGLE_TAXI_LIGHTS",
        "ToggleValueData": "TOGGLE_TAXI_LIGHTS",
        "__sharpDeckUUID": "603ef5812fe64ef59ba108eaf0375545"
      },
      "State": 0,
      "States": [
        {
          "FFamily": "",
          "FSize": "",
          "FStyle": "",
          "FUnderline": "",
          "Image": "state0.png",
          "Title": "",
          "TitleAlignment": "",
          "TitleColor": "",
          "TitleShow": ""
        }
      ],
      "UUID": "tech.flighttracker.streamdeck.generic.toggle"
    }
  },
  "DeviceModel": "20GAT9901",
  "DeviceUUID": "@(1)[4057/108/CL49K2A00889]",
  "Name": "Default Profile",
  "Version": "1.0"
}

Is there any advice on how to fix this problem?

Greetings,
schweito

**Comment by rmroc451** [comment]
> Dear Comunity,
> 
> I am new to the Stream Desk and new to to Flight Tracker plugin - however, I did watch some videos and read the examples. I wanted test a simple Taxi Light toggle function and even though the toggle event works fine, my custom image is not changing as the state of the toggle in the simulator changes. The .json-file generated by the flighttracker-plugin looks like this:
> 
> {
>   "Actions": {
>     "7,0": {
>       "Name": "Generic Toggle Button",
>       "Settings": {
>         "FeedbackValue": "LIGHT TAXI",
>         "ImageOff": "C:/Users/korre/Pictures/EMB-110/Resized/PNG/TAXI_OFF.png",
>         "ImageOff_base64": null,
>         "ImageOn": "C:/Users/korre/Pictures/EMB-110/Resized/PNG/TAXI_ON.png",
>         "ImageOn_base64": null,
>         "ToggleValue": "TOGGLE_TAXI_LIGHTS",
>         "ToggleValueData": "TOGGLE_TAXI_LIGHTS",
>         "__sharpDeckUUID": "603ef5812fe64ef59ba108eaf0375545"
>       },
>       "State": 0,
>       "States": [
>         {
>           "FFamily": "",
>           "FSize": "",
>           "FStyle": "",
>           "FUnderline": "",
>           "Image": "state0.png",
>           "Title": "",
>           "TitleAlignment": "",
>           "TitleColor": "",
>           "TitleShow": ""
>         }
>       ],
>       "UUID": "tech.flighttracker.streamdeck.generic.toggle"
>     }
>   },
>   "DeviceModel": "20GAT9901",
>   "DeviceUUID": "@(1)[4057/108/CL49K2A00889]",
>   "Name": "Default Profile",
>   "Version": "1.0"
> }
> 
> 
> Is there any advice on how to fix this problem?
> 
> Greetings,
> schweito
> 
>  

Can you share a shot of the button configuration screen? I think you might have that "ToggleValueData" node filled in when you don't need it.

When you click the button, does the simulator actually update the taxi light state in the game?

**Reply by schweito** [reply]
Hi,
thanks for the quick reply. Yes, the simulator recognizes the toggle event and flips the switch. Please find the settings attached.

Greetings,
schweito
![flight_tracker_lower_part](https://user-images.githubusercontent.com/96797928/147615418-3e3226a1-c03b-4d29-a3bf-060d942549fb.png)
![flight_tracker_lower_part_II](https://user-images.githubusercontent.com/96797928/147615420-c92a6a40-8a60-42a5-b24c-8c6b71aea966.png)
![flight_tracker_upper_part](https://user-images.githubusercontent.com/96797928/147615421-90243661-e87a-4537-9a63-236b978b36ef.png)
o

**Reply by rmroc451** [reply]
Try LIGHT TAXI ON

**Comment by schweito** [comment]
Hi,
it also did not work it changes the state in the sim but the flight deck does not alternate the images. I am trying to do a profile for the EMB-110 maybe this is the reason why it doesnt work as it is third party? But the toggle event is registered it is just the feedback...and it works with Hotkey-Switches so i guess it has to be simconnect.

When I try the HDG example from the website this also works nicely.
Thanks for your help so far...

Greetings,
schweit
[StreamDeck0.log](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/7788713/StreamDeck0.log)
o

**Reply by nguyenquyhy** [reply]
I'm a bit confused. Which HDG example are you mentioning about?

**Reply by nguyenquyhy** [reply]
The log you attach is Stream Deck log. Can you please attach the log of the plugin instead? Those are *.log files in `%appdata%\Elgato\StreamDeck\Plugins\tech.flighttracker.streamdeck.sdPlugin`

**Reply by schweito** [reply]
Dear Nguyen,
please find the files attached.
Regards,
schweito
[flightstreamdeck20211229.log](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/7792891/flightstreamdeck20211229.log)
[flightstreamdeck20211229_001.log](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/7792892/flightstreamdeck20211229_001.log)

**Reply by rmroc451** [reply]
It looks like you still have LIGHT TAXI vs LIGHT TAXI ON.

After you update the value, click on another field or button on the streamdeck software.

I think the blur isn't firing, which doesn't tell the sim about the steamdeck software about the change you made.

Let us know if that solves the image issue on your buttons.

**Reply by schweito** [reply]
Hi,
according to the logfile there it is shown as taxi light on but it does not sow on the streamdeck. Please find attached the latest logfile.
Regards,
schweito
[flightstreamdeck20211230.log](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/7794747/flightstreamdeck20211230.log)
[StreamDeck0.log](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/7794751/StreamDeck0.log)

**Comment by schweito** [comment]
Hi,
the one on the main page where the buttons and functions are explained. There is an example of diplaying the current heading. The heading bug moves in the plane and is chanhing on the
stream deck as well.
Greetings,
schweito
![example](https://user-images.githubusercontent.com/96797928/147661217-27d67f79-09b3-449c-9b00-d3342331339f.png)
